### PR TITLE
17 observations intervals

### PR DESCRIPTION
**Pull request by BaptisteDlp**
- Allowed values for `tempRes` input in `spatialBehaviour`/`temporalBehaviour` are now: 60, 30 or 15 (minutes).
- + tests
- Minor changes in `getDataInterpolate` so that  it handles every (regular) observations intervals (1h, 2h, 3h etc ...)

### CODE REVIEW

File: R/temporalBehaviour.R [hunk 1]
```diff
@@ -393,8 +393,8 @@ finalizeResult <- function(finalResult, result, product, points, isoT, indices,
 #' Monthly Weather Review, 134(4), 1102–1120. https://doi.org/10.1175/MWR3106.1
 #'
 #' @examples
+#' \dontrun{
```
Comment (thomasarsouze) [inline R/temporalBehaviour.R:396]: In CRAN, they request `\donttest` rather than `\dontrun`

File: R/temporalBehaviour.R [hunk 2]
```diff
@@ -17,42 +17,42 @@ checkInputsTemporalBehaviour <- function(sts, points, product, windThreshold, me
                                          empiricalRMW, tempRes, verbose) {
   # Checking sts input
   stopifnot("no data found" = !missing(sts))
-
+
```
Comment (thomasarsouze) [inline R/temporalBehaviour.R:20]: Linting: remove all trailing whitespaces
```suggestion
  
```

**Review by thomasarsouze — COMMENTED** [review]
Difficult to compare because somehow linting modifications have been done (by your editor ?).

I am not sure to fully understand the modifications here. If I understand correctly you only change the definition of temporal resolution to `minutes` which is great, but wasn't the issue suppposed to be more than just that. Can you confirm @thomaspibanez ?

If this is only that, then I am ok to merge.

**Comment by thomaspibanez** [comment]
I don't really see how these changes address the issue which was not about data interpolation but data importation with different time steps, i.e., not only every 3 or 6 hours but any possible time step (e.g. every 1 hour)?

**Comment by BaptisteDlp** [comment]
The modification are actually in the data interpolation and not importation. We can already import data with different regular time observations

File: R/temporalBehaviour.R (hunk 2 above)
Comment (BaptisteDlp) [reply R/temporalBehaviour.R:20]: I did not understand the pb here

**Comment by BaptisteDlp** [comment]
Here after a reproducible code to test the new feature:

```
timeIntervalInHour = 2 #Change here to simulate a observation interval in the database (in hour)
# Default is 3h or 6h in ibtracks database

tempRes = 15 #Choose a temporal resolution for interpolation, either 60, 30 or 15 minutes


# Simulate new iso times 
df = st@data$PAM@obs.all
time1 = df$iso.time[1]
newIso = c()
for(t in 1:length(df$iso.time)){
  newIso = c(newIso, as.character(as.POSIXct(time1) + timeIntervalInHour*t*60*60))
}
print(newIso)

# Replace new iso times
st@data$PAM@obs.all$iso.time = newIso


TS <- temporalBehaviour(st,
                        points = data.frame(x = c(169), y=c(-15)),
                        tempRes = tempRes)

# Check this temporalBehaviour output and observations each indices/isotimes
print(TS)
```

**Comment by thomasarsouze** [comment]
> The modification are actually in the data interpolation and not importation. We can already import data with different regular time observations

Isn't the point mentioned by @thomaspibanez related to these lines (why are they here by the way ?)
https://github.com/umr-amap/StormR/blob/f5e0cc2f2927bc349514685ba3c7b44444921108/R/defStormsList.R#L872C1-L873C45

**Comment by BaptisteDlp** [comment]
> > The modification are actually in the data interpolation and not importation. We can already import data with different regular time observations
> 
> Isn't the point mentioned by @thomaspibanez related to these lines (why are they here by the way ?) https://github.com/umr-amap/StormR/blob/f5e0cc2f2927bc349514685ba3c7b44444921108/R/defStormsList.R#L872C1-L873C45

Ok I got it,  I forgot to update theses lines

File: R/temporalBehaviour.R (hunk 1 above)
Comment (BaptisteDlp) [reply R/temporalBehaviour.R]: Done

File: R/temporalBehaviour.R (hunk 2 above)
Comment (BaptisteDlp) [reply R/temporalBehaviour.R:20]: I think it's fixed now

**Comment by BaptisteDlp** [comment]
@thomasarsouze  @thomaspibanez ,
Overall, possibles/allowed observation intervals should be 1h, 2h, 3h, 4h, 6h, 8h or 12h, otherwise it cannot be regular and there will be shifts, (Example, a 5h hour observation interval would give, starting from 00h, 5h 10h 15h 20h 1H 6H and so on ...)

Thus, I need to know, How should we handle this in the code ? Should we add an input in the database that specifies the observation interval gap in the database  ? Plus, How  can we be certain that observations always start from 00h in a database ? Seems weird to me if observations could starts at odd numbers? I don't know if you get my point 

I think we need to make things clear and make a minimal set of rules that would dictate the format of the file database

**Comment by thomasarsouze** [comment]
> @thomasarsouze @thomaspibanez , Overall, possibles/allowed observation intervals should be 1h, 2h, 3h, 4h, 6h, 8h or 12h, otherwise it cannot be regular and there will be shifts, (Example, a 5h hour observation interval would give, starting from 00h, 5h 10h 15h 20h 1H 6H and so on ...)

Maybe consider also 24h, but otherwise it's fine.

> 
> Thus, I need to know, How should we handle this in the code ? Should we add an input in the database that specifies the observation interval gap in the database ? 

We should be able to infer that from the time stamps.

> Plus, How can we be certain that observations always start from 00h in a database ? 

This has not to be always the case... Storms will not always start at 00h00. Why do you need that ? Is this something that is done systematicaly in IBTRACKS ?

> Seems weird to me if observations could starts at odd numbers? I don't know if you get my point
> 
> I think we need to make things clear and make a minimal set of rules that would dictate the format of the file database

My suggestion for set of rules would only be:
- infer time interval directly on the dataset, and check it is compliant with the set of possible times you defined above
- we should be able to allow any starting hour, so we don't need to check it.

**Pull request merged**